frame/merkle-mountain-range/primitives/src/lib.rs

```diff
@@ -411,23 +411,23 @@ impl EncodableOpaqueLeaf {
 sp_api::decl_runtime_apis! {
 	/// API to interact with MMR pallet.
 	pub trait MmrApi<Hash: codec::Codec> {
-		/// Generate MMR proof for a leaf under given index.
-		fn generate_proof(leaf_index: LeafIndex) -> Result<(EncodableOpaqueLeaf, Proof<Hash>), Error>;
+		/// Generate MMR proof for leaves under given indexes.
+		fn generate_proof(leaf_indexes: Vec<LeafIndex>) -> Result<(Vec<(LeafIndex, EncodableOpaqueLeaf)>, Proof<Hash>), Error>;
```

Comment:
This is not backwards compatible. I recommend adding new functions instead of updating the old ones.